awx/main/tests/functional/api/test_inventory.py

```diff
@@ -624,6 +624,35 @@ def test_patch_constructed_inventory(self, constructed_inventory, admin_user, pa
         assert inv_src.update_cache_timeout == 54
         assert inv_src.limit == 'foobar'
 
+    def test_patch_constructed_inventory_generated_source_limits_editable_fields(self, constructed_inventory, admin_user, project, patch):
+        inv_src = constructed_inventory.inventory_sources.first()
+        r = patch(
+            url=inv_src.get_absolute_url(),
+            data={
+                'source': 'scm',
+                'source_project': project.pk,
+                'source_path': '',
+                'source_vars': 'plugin: a.b.c',
+            },
+            expect=400,
+            user=admin_user,
+        )
+        assert r.data['error'] == "Cannot change field 'source' on a constructed inventory source."
+
+        # Make sure it didn't get updated before we got the error
+        assert inv_src == constructed_inventory.inventory_sources.first()
```

Comment:
I think this won't do what I want, this probably just compares PK. I should compare the fields themselves.

Reply:
If I do this after the `patch()` it fails, and I have no idea why, especially because if I test _outside_ of pytest (using the collection, getting the 400, and looking at the database), the record doesn't get updated.

```python
inv_src_after_err = constructed_inventory.inventory_sources.first()
assert inv_src.source == inv_src_after_err.source
```